modules/account-lib/src/coin/eth/utils.ts

```diff
@@ -315,6 +315,12 @@ export function decodeFlushTokensData(data: string): FlushTokensData {
  * @returns {TransactionType} The classified transaction type
  */
 export function classifyTransaction(data: string): TransactionType {
+  if (data.length < 10) {
+    // contract calls must have at least 4 bytes (method id) and '0x'
+    // if it doesn't have enough data to be a contract call it must be a single sig send
+    return TransactionType.SingleSigSend;

```

Comment:
this seems pretty hacky. I think this function is weak overall actually. this is really classifying a transaction based on data but not value as well?

Reply:
How do you suggest I do it then? There's no data to classify single sig txs with

Reply:
What do you suggest TT?

Reply:
Why would you classify a transaction based on value? The value doesn't provide any information on the type of transactino we're talking about

Reply:
ok, this seems fine

Reply:
I think my main concern with this function was that it lumps everything that's below a threshold as a single send but this seems to work as it's not a valid contract call period unless it falls below this threshold